lib/Expr/Expr.cpp

```diff
@@ -336,23 +339,33 @@ void Expr::dump() const {
 
 ref<Expr> ConstantExpr::fromMemory(void *address, Width width) {
   switch (width) {
+  default: assert(0 && "invalid width");
   case  Expr::Bool: return ConstantExpr::create(*(( uint8_t*) address), width);
   case  Expr::Int8: return ConstantExpr::create(*(( uint8_t*) address), width);
   case Expr::Int16: return ConstantExpr::create(*((uint16_t*) address), width);
   case Expr::Int32: return ConstantExpr::create(*((uint32_t*) address), width);
   case Expr::Int64: return ConstantExpr::create(*((uint64_t*) address), width);
   // FIXME: what about machines without x87 support?
-  default:
+  case Expr::Fl80:
     return ConstantExpr::alloc(
         llvm::APInt(width,
                     (width + llvm::APFloatBase::integerPartWidth - 1) /
                         llvm::APFloatBase::integerPartWidth,
                     (const uint64_t *)address));
+  case Expr::Int128:
+  case Expr::Int256:
+  case Expr::Int512:
+      return ConstantExpr::alloc(
+              llvm::APInt(width,
```

Comment:
That constructor is marked "deprecated" for ages.

Reply:
Good catch! I just read its description which says:

```
/// Equivalent to APInt(numBits, ArrayRef<uint64_t>(bigVal, numWords)), but
/// deprecated because this constructor is prone to ambiguity with the
/// APInt(unsigned, uint64_t, bool) constructor.
```

Do I need to replace it with `APInt(numBits, ArrayRef<uint64_t>(bigVal, numWords))` call or not?